File: Curso_logica_de_programacao/Exercicios_logica_em_C#/Diagonal_negativos/Diagonal_negativos.cs
using System.Globalization;

namespace Soma_vetor;

class Program
{
    static void Main(string[] args)
    {

        CultureInfo CI = CultureInfo.InvariantCulture;

        int n;

        Console.Write("Qual a ordem da matriz? ");
        n = int.Parse(Console.ReadLine());

        double[,] vet = new double[n, n];

        int negativos = 0;

        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                System.Console.Write("Elemento [" + i + "," + j + "]: ");
                vet[i, j] = double.Parse(Console.ReadLine(), CI);
                if (vet[i, j] < 0)
                {
                    negativos++;
                }
            }
        }


        System.Console.WriteLine("Diagonal principal: ");
        for (int i = 0; i < n; i++)
        {
            System.Console.Write(vet[i, i] + " ");
        }

        Console.WriteLine();

        System.Console.Write("Quantidade de negativos: " + negativos);


    }
}

File: Algoritmos_e_Logica_de_Programacao/Exercicios_logica_em_C++/Diagonal_negativos/Diagonal_negativos.cpp
#include <iomanip>
#include <iostream>

using namespace std;

int main()
{

    int n;

    cout << "Qual a ordem da matriz? ";
    cin >> n;

    int elemento[n][n];
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> elemento[i][j];
        }
    }

    cout << "Diagonal principal: \n";
    for (int i = 0; i < n; i++) {
            cout << elemento[i][i] << " ";
    }

    int NumerosNegativos = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (elemento[i][j] < 0){
                NumerosNegativos++;
            }
        }
    }

    cout << "\nQuantidade de negativos = " << NumerosNegativos;
    
    return 0;
}